internal/rpmmd/repository.go

```diff
@@ -150,6 +156,39 @@ func LoadRepositories(confPaths []string, distro string) (map[string][]RepoConfi
 		return nil, err
 	}
 
+	if strings.HasPrefix(distro, "rhel-8") {
+		var sslClientKey string
+		var sslClientCert string
+
+		regexKey := regexp.MustCompile("^[0-9]*-key.pem")
+		regexCert := regexp.MustCompile("^[0-9]*.pem")
+
+		entitlementFiles, err := ioutil.ReadDir("/etc/pki/entitlement/")
+		if err != nil {
+			return nil, &RepositoryError{"LoadRepositories failed: unable to read /etc/pki/entitlement/"}
+		}
+		for _, file := range entitlementFiles {
+			matchKey := regexKey.MatchString(file.Name())
+			if matchKey {
+				sslClientKey = path.Join("/etc/pki/entitlement/", file.Name())
+				continue
+			}
+
+			matchCert := regexCert.MatchString(file.Name())
+			if matchCert {
+				sslClientCert = path.Join("/etc/pki/entitlement/", file.Name())
+			}
+		}
```

Comment:
This will break if there are multiple entitlements in the directory, because it is not ensuring that the `sslClientCert` and `sslClientKey` are from matching certificates. How about only checking for `regexCert`, and then loading the same file name with `-key` appended?

Also, stop iterating after finding the first one.

Reply:
This is fixed